### PR TITLE
update nvhpc and gnu on derecho

### PR DESCRIPTION
**Pull request by jedwards4b**
Updates gcc, cray and nvhpc compilers on derecho.

### CODE REVIEW

**Review by fischer-ncar — APPROVED** [review]

File: machines/derecho/config_machines.xml [hunk 1]
```diff
@@ -48,20 +48,17 @@
...
+        <command name="load">cray-libsci/23.09.1.1</command>
       </modules>
       <modules compiler="gnu">
-        <command name="load">gcc/12.2.0</command>
+        <command name="load">gcc/13.2.0</command>
         <command name="load">cray-libsci/23.02.1.1</command>
+        <command name="load">openblas/0.3.25</command>
       </modules>
       <modules compiler="nvhpc">
         <command name="load">nvhpc/24.3</command>
```
Comment (sjsprecious) [inline machines/derecho/config_machines.xml:64]: Do you want to update nvhpc to nvhpc/24.7 here?

File: machines/derecho/config_machines.xml [hunk 2]
```diff
@@ -48,20 +48,17 @@
...
-      </modules>
-      <modules compiler="intel-classic">
-        <command name="load">intel-classic/2023.2.1</command>
+        <command name="load">intel-oneapi/2024.0.2</command>
 	<command name="load">mkl</command>
       </modules>
       <modules compiler="cray">
-        <command name="load">cce/15.0.1</command>
-        <command name="load">cray-libsci/23.02.1.1</command>
+        <command name="load">cce/17.0.1</command>
```
Comment (sjsprecious) [inline machines/derecho/config_machines.xml:55]: So does Cray compiler work with CESM now?

Comment (jedwards4b) [reply machines/derecho/config_machines.xml:55]: No - still having issues.

File: machines/derecho/config_machines.xml [hunk 3]
```diff
@@ -91,18 +91,22 @@
 
       <modules DEBUG="FALSE">
 	<command name="load">parallelio/2.6.2</command>
-	<command name="load">esmf/8.6.0</command>
+	<command name="load">esmf/8.6.1</command>
```
Comment (fischer-ncar) [inline machines/derecho/config_machines.xml:94]: esmf/8.6.1 isn't available on derecho.  There is esmf/8.6.1b04.

Comment (jedwards4b) [reply machines/derecho/config_machines.xml:94]:
``` module load intel/2024.0.2

Due to MODULEPATH changes, the following have been reloaded:
  1) cray-mpich/8.1.27     2) netcdf/4.9.2

The following have been reloaded with a version change:
  1) hdf5/1.12.2 => hdf5/1.14.3     2) intel/2023.2.1 => intel/2024.0.2

derecho1: /glade/campaign/cesm/development/espwg/SMYLE-CW3E-L83/cases/b.e21.BSMYLE-CW3E-L83.f09_g17.2013-11.018
:) module avail esmf

----------------------------------------------------------------------------- MPI-dependent Software - [oneapi + cray-mpich] -----------------------------------------------------------------------------
   esmf/8.6.0-debug    esmf/8.6.0    esmf/8.6.1b04    esmf/8.6.1-debug    esmf/8.6.1 (D)

```

Comment (fischer-ncar) [reply machines/derecho/config_machines.xml:94]: esmf/8.6.1 isn't available for intel/2023.2.1, which the prealpha and prebeta tests use.  I do see esmf/8.6.1 is available for intel/2024.0.2 (intel-oneapi).  Are we switching to intel-oneapi?

Comment (jedwards4b) [reply machines/derecho/config_machines.xml:94]: Let me work on this a bit more - I think that we should update to intel/2024.0.2 but not yet to intel-oneapi.

**Review by sjsprecious — APPROVED** [review]

**Pull request merged**